dali/imgcodec/util/convert_test.cc

```diff
@@ -220,6 +235,57 @@ TYPED_TEST(ColorConversionTest, BgrToBgr) {
 }
 
 
+/**
+ * @brief Class for testing Convert's support of different layouts.
+ */
+class ConvertLayoutTest : public ConversionTestBase<float> {
+ public:
+  void Test(const std::string& layout_name, const ROI &roi = {}) {
+    auto rgb_path = GetPath(layout_name, "rgb");
+    auto ycbcr_path = GetPath(layout_name, "ycbcr");
+
+    std::string layout_code = layout_name;
+    for (auto &c : layout_code) c = toupper(c);
+    TensorLayout layout(layout_code);
+
+    auto ref = ReadReferenceFrom(ycbcr_path);
+    if (roi) {
+      ref.SetLayout(layout);
```

Comment:
Why do we set layout only when roi is specified?

Reply:
The layout is only needed by `Crop` and that's how it ended up here. I agree that keeping it inside the `if` is confusing, I moved it out